### ISSUE TITLE
[Bug]: bad function argument + infinite recursion problem

### ISSUE THREAD

**Issue by vladola** [mpenning/ciscoconfparse#262]
### Contact Details

vladola93@gmail.com

### What happened?

Right here:
https://github.com/mpenning/ciscoconfparse/blob/05570afa20b180d73426281acd41e647225631ec/ciscoconfparse/ciscoconfparse.py#L598

There is no argument called 'filename'. It's 'filepath'. The bigger problem is that even if you fix that it creates an infinite recursion loop.

At the moment the library is not usable at all as it fails from initialization. 

### CiscoConfParse Version

1.7.11

### What Operating System are you using?

Microsoft Windows

### What Python version(s) have this problem?

Python 3.9

### Show us how to reproduce the problem.  Please tell us if the problem is specific to certain inputs or situations.

```python
from ciscoconfparse import CiscoConfParse
parser = CiscoConfParser('file.txt')
```

### Python tracebacks

```shell
It first raises a problem with the function not having "filename" as an argument. If you fix that you run into an infinite recursion loop.
```

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/mpenning/ciscoconfparse/blob/main/CONTRIBUTING.md)

**Comment by mpenning** [comment]
`CiscoConfParse()` version 1.7.11 did not read configs from disk correctly.  This happened because I reorganized several things in the `CiscoConfParse()` object init sequence and did not test "read from file" in the test suite.  I think I removed test coverage for this a while back when I was debugging some CI/CD test failures and never added tests for read-from-file back again.

> At the moment the library is not usable at all as it fails from initialization.

That's not true.  There's a workaround for the problem... you can manually read a config filename into a list and pass the list to `CiscoConfParse()`... that said, the test escape for reading a file from disk is very significant and I'm sorry it took almost a day after releasing 1.7.11 to find/fix this problem.

This should be completely fixed in 1.7.13... please test and advise.

**Comment by vladola** [comment]
Thanks for taking care of this so quickly!
